### ISSUE TITLE
[question] A way to remove a dependencies option

### ISSUE THREAD

**Issue by daravi** [conan-io/conan#8542]
I have some packages which use Qt with different options. For example:
pack1: qt::opt1, qt::opt3, qt::opt4
pack2: qt::opt1, qt::opt2

And my end consumer uses these packages as well as Qt with a superset of those options:
consumer: pack1, pack2, qt::opt1, qt::opt2, qt::opt3, qt::opt4, qt::opt5

Is there a way for me to specify in pack1 and pack2 that I do not care about the value of certain options? (instead of indicating that I need the default option)

Otherwise I would have to go back and edit the upstream package whenever I want to add an option to my end consumer.

- [✅] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by solvingj** [comment]
I'm confused by your dilemma.  In theory, you do not indicate that you need a default value for an option.  You don't have to specify anything and you get the default.  

Also, in general, it's much more flexible and less troublesome to specify options dependencies via profiles rather than hard-coding them into recipes of consumers.  It can seem like the right choice at first. For example, if `pack1` ALWAYS should use `qt::opt1=someval`, then you can hardcode that in the `config_options` method of `pack1`.  However, it becomes really difficult to manage and find where options are being declared from.  It becomes tricky to understand what happens when there are conflicts among different recipes in the graph (which always happens as graphs grow).  

In summary, if you create "Complete Profiles" (where all options for all dependencies are declared explicitly) for your "product packages" (packages with no consumers), you will likely find (as have many others) that you never have these awkward and difficult challenges.  Your profiles are nice to read because they declare all the options, and you have ultimate flexibility where any new build can be a new profile, or simply an overridden option at the CLI if it's just a temporary test. 

In summary, in most cases, it's problematic and not very scalable to set options for dependencies inside recipes. Please let me know if this guidance helps.

**Comment by daravi** [comment]
Sorry for delay. I will be adding more details to this issue to explain my use case.

**Comment by daravi** [comment]
@solvingj I use both gammaray and qt in my project. gammaray requires that qt be built with some options like optA and optB. My project requires that qt have optA, optB, optC, optD. When I build my project, conan complains that the options do not match. Is there a way for me to modify the gammaray recipe in such a way to not have to touch it every time I update qt options in my root package? (since gammaray is pretty much ok if I have more options enabled to what it needs and does not care about the value of optC and optD). Is there a way for gammaray to say "I do not care about the value of these options"?

**Comment by daravi** [comment]
Sorry thinking about this more I now realize that my question does not make much sense. Sorry for the noise!

>  It can seem like the right choice at first. For example, if pack1 ALWAYS should use qt::opt1=someval, then you can hardcode that in the config_options method of pack1. However, it becomes really difficult to manage and find where options are being declared from

This was exactly the source of my issue. I removed all options from the packages and instead throw exception in the `validate` method if the option is missing. Would you say this is the right approach?

Thank you and cheers.

**Comment by solvingj** [comment]
Yes, exactly.
